userspace/libsinsp/sinsp.cpp

```diff
@@ -1056,6 +1056,52 @@ uint64_t sinsp::max_buf_used()
 	}
 }
 
+void sinsp::get_procs_cpu_from_driver(uint64_t ts)
+{
+	struct timeval tod;
+
+	if(ts <= m_next_flush_time_ns)
+	{
+		return;
+	}
+
+	uint64_t next_full_second = ts - (ts % ONE_SECOND_IN_NS) + ONE_SECOND_IN_NS;
+
+	if(m_next_flush_time_ns == 0)
+	{
+		m_next_flush_time_ns = next_full_second;
+		return;
+	}
+
+	m_next_flush_time_ns = next_full_second;
+
+	gettimeofday(&tod, NULL);
+	uint64_t procrequest_tod = (uint64_t)tod.tv_sec * 1000000000 + tod.tv_usec * 1000;
```

Comment:
This code basically just reimplements `sinsp_utils::get_current_time_ns()`...maybe you can clean that up while you're in here?